File: Projeto-TCC/ACS-Cadastro-Mobile/app/src/main/java/tcc/acs_cadastro_mobile/persistence/SocialDemographicPersistence.java
package tcc.acs_cadastro_mobile.persistence;

import io.realm.Realm;
import tcc.acs_cadastro_mobile.models.SocialDemographicModel;
import tcc.acs_cadastro_mobile.subModels.CommunityTraditional;
import tcc.acs_cadastro_mobile.subModels.Deficiency;
import tcc.acs_cadastro_mobile.subModels.EducationEmployment;
import tcc.acs_cadastro_mobile.subModels.HealthAndGroup;
import tcc.acs_cadastro_mobile.subModels.SexualOrientation;

public class SocialDemographicPersistence {

    private SocialDemographicPersistence() {}

    public static SocialDemographicModel get(String kinship, EducationEmployment educationEmployment,
                                             HealthAndGroup healthAndGroup, String kids09, CommunityTraditional communityTraditional,
                                             SexualOrientation sexualOrientation, Deficiency deficiency){
        Realm realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        SocialDemographicModel object = realm.copyToRealm(new SocialDemographicModel(kinship, educationEmployment,
                healthAndGroup, kids09, communityTraditional, sexualOrientation, deficiency));
        realm.commitTransaction();
        realm.close();
        return object;
    }

    public static EducationEmployment get(boolean school, String occupation, String education, String employment){

        Realm realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        EducationEmployment object = realm.copyToRealm(new EducationEmployment(school, occupation, education, employment));
        realm.commitTransaction();
        realm.close();
        return object;
    }

    public static HealthAndGroup get(boolean caregiver, boolean communityGroup, boolean healthPlan){
        Realm realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        HealthAndGroup object = realm.copyToRealm(new HealthAndGroup(caregiver, communityGroup, healthPlan));
        realm.commitTransaction();
        realm.close();
        return object;
    }

    public static CommunityTraditional getCommunityTraditional(boolean isCommunityTraditional, String value){
        Realm realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        CommunityTraditional object = realm.copyToRealm(new CommunityTraditional(isCommunityTraditional, value));
        realm.commitTransaction();
        realm.close();
        return object;
    }

    public static SexualOrientation getSexualOrientation(boolean isOrientation, String value){

        Realm realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        SexualOrientation object = realm.copyToRealm(new SexualOrientation(isOrientation, value));
        realm.commitTransaction();
        realm.close();
        return object;
    }

    public static Deficiency get(boolean isDeficiency, boolean[] deficiency){
        Realm realm = Realm.getDefaultInstance();
        realm.beginTransaction();
        Deficiency object = realm.copyToRealm(new Deficiency(isDeficiency, deficiency));
        realm.commitTransaction();
        realm.close();
        return object;
    }
}

File: Projeto-TCC/ACS-Cadastro-Web/persistences/SocialDemographicPersistence.php
<?php

class SocialDemographicPersistence
{
    public static function getById(AcsDataBase $db, $id)
    {
        $query = "SELECT SD.KINSHIP, SD.KIDS_09,
                    D.DEFICIENCY, D.HEARING, D.VISUAL, D.INTELLECTUAL, D.PHYSICAL, D.ANOTHER,
                    S.SEXUAL_ORIENTATION, S.DESCRIPTION AS SEX_DESCRIPTION,
                    C.COMMUNITY_TRADITIONAL, C.DESCRIPTION AS COMM_DESCRIPTION,
                    H.CAREGIVER, H.COMMUNITY_GROUP, H.HEALTH_PLAN,
                    E.SCHOOL, E.OCCUPATION, E.EDUCATION, E.EMPLOYMENT
                FROM tb_social_demographic SD
                INNER JOIN tb_deficiency AS D ON SD.ID_DEFICIENCY = D.DEFICIENCY_ID
                INNER JOIN tb_sexual_orientation AS S ON SD.ID_SEXUAL_ORIENTATION = S.SEXUAL_ORIENTATION_ID
                INNER JOIN tb_community_traditional AS C ON SD.ID_COMMUNITY_TRADITIONAL = C.COMMUNITY_TRADITIONAL_ID
                INNER JOIN tb_health_group AS H ON SD.ID_HEALTH_GROUP = H.HEALTH_GROUP_ID
                INNER JOIN tb_education_employment AS E ON SD.ID_EDUCATION_EMPLOYMENT = E.EDUCATION_EMPLOYMENT_ID
                WHERE SD.SOCIAL_DEMOGRAPHIC_ID = :SOCIAL_DEMOGRAPHIC";

        return $db->select($query, array(":" . SocialDemographicModel::SOCIAL_DEMOGRAPHIC => $id));
    }
    static function insert(AcsDataBase $db, SocialDemographicModel $socialDemographic)
    {
        $values = $socialDemographic->getValuesToDB();
        $ids[":ID_" . EducationEmployment::EDUCATION_EMPLOYMENT] = $db->insert(self::queryEducationEmployment(), $values[EducationEmployment::EDUCATION_EMPLOYMENT]);
        $ids[":ID_" . HealthGroup::HEALTH_GROUP] = $db->insert(self::queryHealthGroup(), $values[HealthGroup::HEALTH_GROUP]);
        $ids[":ID_" . CommunityTraditional::COMMUNITY_TRADITIONAL] = $db->insert(self::queryCommunityTraditional(), $values[CommunityTraditional::COMMUNITY_TRADITIONAL]);
        $ids[":ID_" . SexualOrientation::SEXUAL_ORIENTATION ] = $db->insert(self::querySexualOrientation(), $values[SexualOrientation::SEXUAL_ORIENTATION ]);
        $ids[":ID_" . Deficiency::DEFICIENCY] = $db->insert(self::queryDeficiency(), $values[Deficiency::DEFICIENCY]);

        if(in_array(false, $ids)){
            return false;
        }
        $ids[":" . SocialDemographicModel::KIDS_09] = $values[SocialDemographicModel::KIDS_09];
        $ids[":" . SocialDemographicModel::KINSHIP] = $values[SocialDemographicModel::KINSHIP];
        return $db->insert(self::querySocialDemographic(), $ids);
    }

    private static function querySocialDemographic()
    {
        return "INSERT INTO TB_SOCIAL_DEMOGRAPHIC (KINSHIP, KIDS_09, ID_DEFICIENCY, ID_SEXUAL_ORIENTATION, 
                  ID_COMMUNITY_TRADITIONAL, ID_HEALTH_GROUP, ID_EDUCATION_EMPLOYMENT) 
                VALUES(:KINSHIP, :KIDS_09, :ID_DEFICIENCY, :ID_SEXUAL_ORIENTATION, 
                  :ID_COMMUNITY_TRADITIONAL, :ID_HEALTH_GROUP, :ID_EDUCATION_EMPLOYMENT)";
    }

    private static function queryEducationEmployment()
    {
        return "INSERT INTO TB_EDUCATION_EMPLOYMENT(SCHOOL, OCCUPATION, EDUCATION, EMPLOYMENT)
                  VALUES(:SCHOOL, :OCCUPATION, :EDUCATION, :EMPLOYMENT)";
    }

    private static function queryHealthGroup()
    {
        return "INSERT INTO TB_HEALTH_GROUP (CAREGIVER, COMMUNITY_GROUP, HEALTH_PLAN) 
                  VALUES (:CAREGIVER, :COMMUNITY_GROUP, :HEALTH_PLAN)";
    }

    private static function queryCommunityTraditional()
    {
        return "INSERT INTO TB_COMMUNITY_TRADITIONAL(COMMUNITY_TRADITIONAL, DESCRIPTION) 
                  VALUES (:COMMUNITY_TRADITIONAL, :DESCRIPTION)";
    }

    private static function querySexualOrientation()
    {
        return "INSERT INTO TB_SEXUAL_ORIENTATION(SEXUAL_ORIENTATION, DESCRIPTION) 
                  VALUES (:SEXUAL_ORIENTATION, :DESCRIPTION)";
    }

    private static function queryDeficiency()
    {
        return "INSERT INTO TB_DEFICIENCY(DEFICIENCY, HEARING, VISUAL, INTELLECTUAL, PHYSICAL, ANOTHER) 
                  VALUES (:DEFICIENCY, :HEARING, :VISUAL, :INTELLECTUAL, :PHYSICAL, :ANOTHER)";
    }
}